content/opt/setup/icinga2

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+if [ ! "$(ls -A /etc/icinga2)" ]; then
+  echo "=> Copying fresh config-files for /etc/icinga2"
+  cp -R /etc/icinga2.dist/* /etc/icinga2/
+fi
+
+# chown directories and files that might be coming from volumes
+mkdir -p /var/log/icinga2/compat/archives
+chown -R nagios:adm /var/log/icinga2
+
+mkdir -p /var/lib/icinga2/api/zones
+mkdir -p /var/lib/icinga2/api/log
+mkdir -p /var/lib/icinga2/api/repository
+chown -R nagios:nagios /var/lib/icinga2
+
+chown -R nagios:root /etc/icinga2
+
+# enable neccessary features
+icinga2 feature enable ido-mysql livestatus compatlog command
+
+#icinga2 API cert - regenerate new private key and certificate when running in a new container
+if [ ! -f /etc/icinga2/pki/$(hostname).key ]; then
+	icinga2 node setup --master
+fi
+
+chfn -f "${ICINGA2_USER_FULLNAME}" nagios
+
+# MySQL installation
+mysql <<-END
+  CREATE DATABASE IF NOT EXISTS icinga;
+  GRANT SELECT, INSERT, UPDATE, DELETE, DROP, CREATE VIEW, INDEX, EXECUTE ON icinga.* TO 'icinga'@'localhost' IDENTIFIED BY '${ICINGA_PASSWORD}';
+END
+
+mysql icinga < /usr/share/icinga2-ido-mysql/schema/mysql.sql >> /var/log/icinga2/mysql-schema.log 2>&1
```

Comment:
The key thought on this change had been:

On a docker-container you're able to configure your stuff via `ENV`-variables. And you save the stuff via volumes. So during boot, the docker-container should process the variables and change the volume data to represent the ENV-settings. The current master uses a logfile approach and changes everything on first container boot and touches a lockfile.

While volumes are persistent over container creation, `ENV`s are not. This leads to a new problem: If you change your ENVs, the volume data does not change automatically. So, my intention is to run the complete setup on every boot, but only **change neccessary values** without overwriting other stuff (which may be created manually by the user). This is why, I removed the sed-calls with `ini_set`, etc.

I created this initial note, to remember, that I haven't tested this actively. It worked, but I didn't know if it worked correctly.

As you mentioned, schema upgrades are also another topic, which this image still requires manual interaction.